Feature/data download

antimaLinux opened the pull request:
## ADD
- added download feature

antimaLinux committed:
[kickest][add] download feature

antimaLinux committed:
[kickest][add] download return

antimaLinux committed:
[kickest][add] moved check variable

antimaLinux committed:
[kickest][change] reverted move

antimaLinux committed:
[kickest][fix] fixed type annotations

antimaLinux committed:
[kickest][fix] fixed trailing line + updated travis conf

antimaLinux committed:
[kickest][fix] removed unused import

antimaLinux committed:
[kickest][fix] edit travis conf for pytest

antimaLinux committed:
[kickest][fix] disabled pytype check for splinter

drugilsberg commented on .flake8:
```diff
@@ -0,0 +1,23 @@
+# W503 line break occurred before a binary operator (badly implemented, doesn't allow log operators within brackets)
+
+
+ignore = E114,E116,E203,E221,E222,E241,E251,E272,E722,E731,W503
```
maybe add newline

drugilsberg commented on kickeststats/helpers/data.py:
```diff
@@ -0,0 +1,10 @@
+from typing import Iterable, Any
+
+
+def grouper(iterable: Iterable, n: int, fillvalue: Any = None) -> Iterable:
```
classic grouper :)

antimaLinux replied:
yes! itertools recipe! why don't they put the recipes in the package? why? :(

drugilsberg reviewed (commented):
Will pull and change the single to double quotes also in teh remaining files.

drugilsberg committed:
feat: syle and installer fix.

Added installable script for data download.

Getting chromedriver from resources.

Signed-off-by: Matteo Manica <drugilsberg@gmail.com>

drugilsberg committed:
docs: improved docs for data download.

Signed-off-by: Matteo Manica <drugilsberg@gmail.com>

drugilsberg committed:
feat: latest list of players.

Signed-off-by: Matteo Manica <drugilsberg@gmail.com>

drugilsberg commented on kickeststats/constants.py:
```diff
@@ -0,0 +1,18 @@
+    )
+)
+
+if not CHROMEDRIVER_EXECUTABLE_PATH:
```
not needed anymore, but we can keep it

drugilsberg commented on kickeststats/download.py:
```diff
@@ -0,0 +1,112 @@
+from typing import List
+import time
+
+from .helpers.parsers import HeaderParser, RowParser, PaginationParser
```
changed to relative imports

drugilsberg commented on setup.py:
```diff
@@ -5,20 +5,25 @@
         ]
     },
-    install_requires=['pandas==1.1.2']
+    install_requires=[
```
added dependencies here for installation from PYPI.

drugilsberg committed:
ci: adding test for script installation.

Signed-off-by: Matteo Manica <drugilsberg@gmail.com>

antimaLinux committed:
[kickest][fix] headless + incognito

drugilsberg committed:
feat: improving player parser from dict.

Signed-off-by: Matteo Manica <drugilsberg@gmail.com>

drugilsberg committed:
Merge branch 'feature/data_download' of github.com:antimaLinux/kickscraper into feature/data_download

drugilsberg committed:
docs: updating usage description.

Signed-off-by: Matteo Manica <drugilsberg@gmail.com>

antimaLinux committed:
[kickest][fix] conversion

antimaLinux committed:
Merge branch 'feature/data_download' of https://github.com/antimaLinux/kickeststats into feature/data_download

antimaLinux committed:
[kickest][fix] required package

drugilsberg committed:
style: fixing typing for mypy.

Signed-off-by: Matteo Manica <drugilsberg@gmail.com>
